File: Java/Basics To Java/src/D/D4.java
package D;

import java.util.Scanner;

public class D4 {
	public static void mainx(String[] args) {
		Scanner sc = new Scanner(System.in);
		int max, vyska;
		String nejvyssi, jmeno;

		System.out.println("zadej jméno žáka. 0 = konec");
		jmeno = sc.next();
		System.out.println("zadej výšku žáka.");
		vyska = sc.nextInt();
		max = vyska;
		nejvyssi = jmeno;

		do {
			System.out.println("zadej jméno žáka. 0 = konec");
			jmeno = sc.next();
			if (!jmeno.equals("0")) {
				System.out.println("zadej výšku žáka.");
				vyska = sc.nextInt();

				if (vyska > max) {
					max = vyska;
					nejvyssi = jmeno;
				}
			}
		} while (!jmeno.equals("0"));
		System.out.println("nejvyšší žák je: " + nejvyssi + "\n" + "výška: " + max);
	}
}

File: C#/Basic logic/D/D4.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace D
{
    internal class D4
    {
        static void Mainx(string[] args)
        {
            int max, vyska;
            string nejvyssi, jmeno;

            Console.WriteLine("zadej jméno žáka. prázné jméno = konec");
            jmeno = Console.ReadLine();
            Console.WriteLine("zadej výšku žáka.");
            vyska = int.Parse(Console.ReadLine());
            max = vyska;
            nejvyssi = jmeno;

            do
            {
                Console.WriteLine("zadej jméno žáka. prázné jméno = konec");
                jmeno = Console.ReadLine();
                if (jmeno != "")
                {
                    Console.WriteLine("zadej výšku žáka.");
                    vyska = int.Parse(Console.ReadLine());

                    if (vyska > max)
                    {
                        max = vyska;
                        nejvyssi = jmeno;
                    }
                }
            }
            while (jmeno != "");
            Console.WriteLine("nejvyšší žák je: " + nejvyssi + "\n" + "výška: " + max);

        }
    }
}
